.github/workflows/e2e_regression.yml

```diff
@@ -39,82 +24,27 @@ jobs:
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Login to Azure
-        if: ${{ !matrix.platform.self-hosted }}
-        uses: azure/login@a65d910e8af852a8061c627c456678983e180302 # v2.2.0
-        with:
-          creds: ${{ secrets.CONTRAST_CI_INFRA_AZURE }}
-      - uses: nicknovitski/nix-develop@a2060d116a50b36dfab02280af558e73ab52427d # v1.1.0
-      - name: Create justfile.env
-        run: |
-          cat <<EOF > justfile.env
-          container_registry=${{ env.container_registry }}
-          azure_resource_group=${{ env.azure_resource_group }}
-          EOF
-      - name: Get credentials for CI cluster
-        if: ${{ !matrix.platform.self-hosted }}
-        run: |
-          just get-credentials
-      - if: ${{ !matrix.platform.self-hosted }}
-        name: Set sync environment
-        run: |
-          sync_ip=$(kubectl get svc sync -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
-          echo "SYNC_ENDPOINT=http://$sync_ip:8080" | tee -a "$GITHUB_ENV"
-          sync_uuid=$(kubectl get configmap sync-server-fifo -o jsonpath='{.data.uuid}')
-          echo "SYNC_FIFO_UUID=$sync_uuid" | tee -a "$GITHUB_ENV"
-      - name: Build and prepare deployments
-        run: |
-          just coordinator initializer port-forwarder openssl service-mesh-proxy node-installer ${{ matrix.platform.name }}
-      - name: Run regression test
-        run: |
-          nix run .#scripts.get-logs workspace/e2e.namespace &
-          nix shell -L .#contrast.e2e --command ${{ matrix.case }}.test -test.v \
-            --image-replacements workspace/just.containerlookup \
-            --namespace-file workspace/e2e.namespace \
-            --platform ${{ matrix.platform.name }} \
-            --skip-undeploy="${{ inputs.skip-undeploy && 'true' || 'false' }}"
-      - name: Upload logs
-        if: always()
-        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0
-        with:
-          name: e2e_pod_logs-${{ matrix.platform.name }}-${{ matrix.case }}
-          path: workspace/namespace-logs
-      - name: Notify teams channel of failure
-        if: ${{ failure() && github.ref == 'main' }}
-        uses: ./.github/actions/post_to_teams
-        with:
-          webhook: ${{ secrets.TEAMS_CI_WEBHOOK }}
-          message: "Regression test failed"
-      - name: Cleanup
-        if: cancelled() && !inputs.skip-undeploy
-        run: |
-          kubectl delete ns "$(cat workspace/e2e.namespace)" --timeout 5m
+    name: "${{ matrix.platform.name }}"
+    uses: ./.github/workflows/e2e.yaml
+    with:
+      skip-undeploy: false
+      test-name: ${{ matrix.test-name }}
+      platform: ${{ matrix.platform.name }}
+      runner: ${{ matrix.platform.runner }}
+      self-hosted: ${{ matrix.platform.self-hosted }}
+      send-failure-notifications: true
```

Comment:
Should only send notifications if running on schedule, right?

Reply:
Yup, good catch. I removed `send-failure-notifications` in favor of just always checking for `github.event_name == 'schedule'`.